packages/host/app/components/matrix/room-message.gts

```diff
@@ -40,24 +40,55 @@ interface Signature {
     isStreaming: boolean;
     currentEditor: number | undefined;
     setCurrentEditor: (editor: number | undefined) => void;
+    retryAction?: () => void;
     isPending?: boolean;
   };
 }
 
-export default class Room extends Component<Signature> {
+export default class RoomMessage extends Component<Signature> {
+  constructor(owner: unknown, args: Signature['Args']) {
+    super(owner, args);
+
+    this.checkStreamingTimeout.perform();
+  }
+
+  @tracked streamingTimeout = false;
+
+  checkStreamingTimeout = task(async () => {
+    if (!this.isFromAssistant || !this.args.isStreaming) {
+      return;
+    }
+
+    // If message is streaming and hasn't been updated in the last minute, show a timeout message
+    if (Date.now() - Number(this.args.message.updated) > 60000) {
+      this.streamingTimeout = true;
+      return;
+    }
+
+    // Do this check every second
+    await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
There’s a [`timeout` built into](https://ember-concurrency.com/api/global.html#timeout) Ember Concurrency, maybe useful here?

Reply:
Yeah that's cleaner. I adjusted it

Reply:
Actually, I had to revert because if I use `timeout`, the runloop will make the test wait until the polling (i.e the task) is completed while the test tries to assert on state that happen *during* polling.

Reply:
oh yes, sad but makes sense